index.js

```diff
@@ -1,125 +1,201 @@
-        const freezeNamespaceObject = localName => {
-          return t.expressionStatement(
-            t.callExpression(
-              t.memberExpression(t.identifier('Object'), t.identifier('freeze')),
-              [t.identifier(localName)]))
+        if (!pattern.startsWith('.')) {
+          throw error(`Glob pattern must be relative, was '${pattern}'`)
         }
 
-        const replacement = node.specifiers.map(specifier => {
-          const localName = specifier.local.name
-          if (specifier.type === 'ImportSpecifier') {
-            const importName = specifier.imported.name
-            if (!lookup[importName]) {
-              throw path.buildCodeFrameError(
-                `Could not match import '${importName}' to a module. Available members are '${members.join("', '")}'`)
-            }
+        const cwd = dirname(state.file.opts.filename)
+        const gm = GlobSync(pattern, {cwd, strict: true})
+        const members = generateMembers(gm, cwd)
+        const unique = Object.create(null)
+        members.forEach(m => {
+          if (m.name === null) {
+            throw error(`Could not generate a valid identifier for '${m.file}'`)
+          }
+          if (unique[m.name]) {
+            // hyphen conversion means foo-bar and fooBar will collide.
+            throw error(`Found colliding members '${m.name}'`)
+          }
+          unique[m.name] = true
+        })
 
-            return makeImport(localName, lookup[importName])
+        const replacement = []
+        specifiers.forEach(s => {
+          const type = s.type
+          const localName = s.local.name
+          if (type === 'ImportSpecifier') {
+            const importName = s.imported.name
+            const member = members.find(m => m.name === importName)
+            if (!member) {
+              const names = members.map(m => m.name).join("', '")
+              throw error(`Could not match import '${importName}' to a module. Available members are '${names}'`)
+            }
+            replacement.push(makeImport(t, localName, member.relative))
+          } else {
+            // Only ImportNamespaceSpecifier can be remaining, since
+            // importDefaultSpecifier has previously been rejected.
+            members.forEach(
+              m => replacement.push(
+                  makeImport(t, `_${localName}_${m.name}`, m.relative)
+              )
+            )
+            replacement.push(
+              makeNamespaceObject(t, localName, members),
+              freezeNamespaceObject(t, localName)
+            )
           }
+        })
 
-          // Only ImportNamespaceSpecifier can be remaining, since
-          // importDefaultSpecifier has previously been rejected.
-          return [].concat(
-            members.map(member => makeImport(`_${localName}_${member}`, lookup[member])),
-            makeNamespaceObject(localName, members),
-            freezeNamespaceObject(localName)
+        if (replacement.length === 0) {
```

Comment:
That's cool, I was trying to highlight that checking `replacement.length` is an indirect test for `specifiers.length === 0`, so we could just have an `if`/`else` condition on that.